runtime/v2/runc/manager/manager_linux.go

```diff
@@ -270,8 +270,10 @@ func (manager) Stop(ctx context.Context, id string) (shim.StopStatus, error) {
 	if err := mount.UnmountAll(filepath.Join(path, "rootfs"), 0); err != nil {
 		log.G(ctx).WithError(err).Warn("failed to cleanup rootfs mount")
 	}
+	pid, _ := runcC.ReadPidFile(filepath.Join(path, process.InitPidFile))
```

Comment:
Shouldn't it return err to the caller? It would be an edge case, but removing the PID file is possible in theory.

Reply:
This line is refers to shim v1's implementation
https://github.com/containerd/containerd/blob/60703af9fdf113eb41520df7f5f17e5e831a83c1/runtime/v1/linux/runtime.go#L449

Normally, the PID file is exist unless remove it manually.  If the PID file remove it unexpectly, `runcC.ReadPidFile` will return -1. In this case there is also no side effect. Check the err and give a warn log here if read PID file failed ?

https://github.com/containerd/go-runc/blob/1c99ade6c04ed41bc4c9a8ac12b81b5fd4e14bcc/utils.go#L32

Reply:
How about returning the error instead of logging? The function should return either a valid value or error.